src/dotnet/OpenAPI.Validator.Tests/OpenAPIModelerValidationTests.cs

```diff
@@ -890,6 +890,23 @@ public void ValidExtensionForLongRunningOperationOptionsForLROPostWithNoResponse
             var messages = GetValidationMessagesForRule<LongRunningOperationsOptionsValidator>(Path.Combine("positive", "long-running-operations-options-positive-4.json"));
             Assert.Empty(messages);
         }
+        
+        [Fact]
+        public void ValidXMSPathWithOAata() 
+        {
+            var filePath = Path.Combine(AutoRest.Core.Utilities.Extensions.CodeBaseDirectory, "Resource", "OpenAPI", "Validation", "positive", "x-ms-paths-with-odata.json");
+            var fileText = File.ReadAllText(filePath);
+            var message = "";
+            try
+            {
+                var servDef = SwaggerParser.Parse(filePath, fileText);
```

Comment:
this tests if the SwaggerParser can Parse src/dotnet/OpenAPI.Validator.Tests/Resource/OpenAPI/Validation/positive/x-ms-paths-with-odata.json correctly.